internal/sync/lifecycle.go

```diff
@@ -182,6 +184,16 @@ func (l *lifecycleOnce) Stop(f func() error) error {
 	return l.loadError()
 }
 
+// Started returns a channel that will close when the lifecycle starts.
+func (l *lifecycleOnce) Started() <-chan struct{} {
```

Comment:
Unfortunately if we do that we can't encapsulate the channel in a switch statement (allowing us to wait for start, or until the context times out).  Since they are directed channels I don't know if people are able to close them, and since this is still an internal API it would probably be one of us shooting ourselves in the foot.